### PR TITLE
Move stage frames to Varyings

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/19038/Move-lighting-background-haze-bloom-stage-frames-to-Varyings

No functional change, just reorganizing some stuff at @samcake's request.

Test plan:
- Verify that bloom, haze, skyboxes, local lights, and shadows still work.  Play around with zones and verify that you see the effects as expected.  Try all the modes: inherit, disabled, enabled.
- On Android, of those effects, only skyboxes are enabled.  Verify that they work.

### CODE REVIEW

File: libraries/render-utils/src/DeferredLightingEffect.cpp [hunk 1]
```diff
@@ -70,17 +70,22 @@ void DeferredLightingEffect::init() {
     loadLightProgram(shader::render_utils::program::local_lights_drawOutline, true, _localLightOutline, _localLightOutlineLocations);
 }
 
+// FIXME: figure out how to move lightFrame into a varying in GeometryCache and RenderPipelines
 void DeferredLightingEffect::setupKeyLightBatch(const RenderArgs* args, gpu::Batch& batch) {
+    setupKeyLightBatch(args, batch, args->_scene->getStage<LightStage>()->_currentFrame);
```
Comment (SamGondelman) [inline libraries/render-utils/src/DeferredLightingEffect.cpp]: This is the last place where we use _currentFrame outside of the zone/light rendering and the stage setup.  I'm not sure how best to remove it.  Maybe cache it during a different stage and use the cached version here?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13515/HighFidelity-Beta-PR14119-de90ce5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13515/HighFidelity-Beta-PR14119-de90ce5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13515/HighFidelity-Beta-Interface-PR14119-de90ce5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13515/HighFidelity-Beta-PR14119-de90ce5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13515/HighFidelity-Beta-Interface-PR14119-de90ce5.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13737/HighFidelity-Beta-PR14119-c10bb7d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13737/HighFidelity-Beta-PR14119-c10bb7d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13737/HighFidelity-Beta-Interface-PR14119-c10bb7d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13737/HighFidelity-Beta-PR14119-c10bb7d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13737/HighFidelity-Beta-Interface-PR14119-c10bb7d.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13983/HighFidelity-Beta-PR14119-c10bb7d.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13983/HighFidelity-Beta-PR14119-c10bb7d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13983/HighFidelity-Beta-Interface-PR14119-c10bb7d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13983/HighFidelity-Beta-PR14119-c10bb7d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13983/HighFidelity-Beta-Interface-PR14119-c10bb7d.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Pull request merged**